src/ansiblelint/config.py

```diff
@@ -205,6 +205,14 @@ def in_venv() -> bool:
 def guess_install_method() -> str:
     """Guess if pip upgrade command should be used."""
     package_name = "ansible-lint"
+
+    try:
+        if distribution(package_name).read_text("INSTALLER").strip() != "pip":  # type: ignore[union-attr]
```

Comment:
```
         if distribution(package_name).read_text("INSTALLER").strip() != "pip":  # type: ignore[union-attr]
```

This returns None if the file doesn't exist which can happen, as I don't think INSTALLER is a mandatory part of the dist-info directory, so calling .strip()` directly is incorrect. Mypy warns about this for a reason...

You can probably change this to

```
        if (distribution(package_name).read_text("INSTALLER") or "").strip() != "pip":
```

Reply:
Updated in PR https://github.com/ansible/ansible-lint/pull/3553